File: food-donation-frontend/src/services/profileService.js
import apiClient from './apiClient';

const profileService = {

  // ─── Basic Profile (instant) ──────────────────────────────
  updateBasicProfile: async (payload) => {
    const { data } = await apiClient.put('/profile/basic', payload);
    return data.data;
  },

  // ─── Submit change request ────────────────────────────────
  submitContactChangeRequest: async (payload) => {
    const { data } = await apiClient.post('/profile/change-request', payload);
    return data.data;
  },

  getMyChangeRequests: async (params = {}) => {
    const { data } = await apiClient.get('/profile/change-requests/my', { params });
    return data.data;
  },

  /**
   * User clicks "Send OTP" button — only works if request is APPROVED.
   * OTP is NOT sent automatically on admin approval.
   */
  sendOtp: async (requestId) => {
    const { data } = await apiClient.post(`/profile/change-requests/${requestId}/send-otp`);
    return data;
  },

  verifyOtp: async (payload) => {
    const { data } = await apiClient.post('/profile/verify-otp', payload);
    return data.data;
  },

  // ─── Admin ────────────────────────────────────────────────
  getAllChangeRequests: async (params = {}) => {
    const { data } = await apiClient.get('/profile/admin/change-requests', { params });
    return data.data;
  },

  approveChangeRequest: async (id, adminNote) => {
    const { data } = await apiClient.put(
      `/profile/admin/change-requests/${id}/approve`,
      { adminNote }
    );
    return data.data;
  },

  rejectChangeRequest: async (id, adminNote) => {
    const { data } = await apiClient.put(
      `/profile/admin/change-requests/${id}/reject`,
      { adminNote }
    );
    return data.data;
  },
};

export default profileService;

File: food-donation-backend/src/main/java/com/fooddonation/service/ProfileService.java
package com.fooddonation.service;

import org.springframework.data.domain.Pageable;

import com.fooddonation.dto.request.*;
import com.fooddonation.dto.response.*;
import com.fooddonation.enums.RequestStatus;

public interface ProfileService {

	/**
	 * Update non-sensitive fields instantly (name, address, city, state, pincode)
	 */
	UserResponse updateBasicProfile(String email, UpdateBasicProfileRequest request);

	/** Submit a request to change email or phone — goes to admin for approval */
	ProfileChangeRequestResponse submitContactChangeRequest(String email, ContactChangeRequest request);

	/** Get all change requests for the current user */
	PageResponse<ProfileChangeRequestResponse> getMyChangeRequests(String email, Pageable pageable);

	/**
	 * Send OTP to the new email/phone. Called ONLY when the user explicitly clicks
	 * "Send OTP" after admin approval. Status must be APPROVED or OTP_SENT (resend
	 * case).
	 */
	void sendOtpToUser(Long requestId, String userEmail);

	/** Verify OTP entered by user and apply the contact change */
	UserResponse verifyOtpAndApplyChange(String email, VerifyOtpRequest request);

	// ─── Admin ───────────────────────────────────────────────

	/** Admin — Get all change requests (optionally filtered by status) */
	PageResponse<ProfileChangeRequestResponse> getAllChangeRequests(RequestStatus status, Pageable pageable);

	/** Admin approves — status → APPROVED. OTP is NOT sent here. */
	ProfileChangeRequestResponse approveChangeRequest(Long requestId, String adminNote, String adminEmail);

	/** Admin rejects a request */
	ProfileChangeRequestResponse rejectChangeRequest(Long requestId, String adminNote, String adminEmail);
}
